core/iwasm/include/wasm_c_api.h

```diff
@@ -527,16 +527,20 @@ typedef struct WASMModuleCommon *wasm_module_t;
 #define LOAD_ARGS_OPTION_DEFINED
 typedef struct LoadArgs {
     char *name;
+    /* True by default.
+    If false, the wasm input buffer (wasm_byte_vec_t) is referenced by the
+    module instead of being cloned. Hence, it should be freed exactly before
+    calling wasm_func_call and not before, as other Wasm C APIs use it. */
+    bool clone_wasm_binary;
     /* TODO: more fields? */
 } LoadArgs;
 #endif /* LOAD_ARGS_OPTION_DEFINED */
 
-WASM_API_EXTERN own wasm_module_t* wasm_module_new(
-  wasm_store_t*, const wasm_byte_vec_t* binary);
+WASM_API_EXTERN own wasm_module_t* wasm_module_new(wasm_store_t*, wasm_byte_vec_t* binary);
```

Comment:
please avoid dropping const for wasm_module_new.

Reply:
I can do that and then use a cast internally from `wasm_module_new`:
```c
module_ex->binary = (wasm_byte_vec_t *)binary;
```
In that way, the external API still has `const` that shows it doesn't modify the binary, that I think it's your concern.

Reply:
Agree to keep const flag for this function, and in wasm_module_new, convert it non-const:
```C
    return wasm_module_new_ex(store, (wasm_byte_vec_t *)binary, &args);
```

Reply:
The cast there wouldn't work if we want to keep const for `wasm_module_new_ex` too

Reply:
Oh, my understanding is that we want to remove `const` for wasm_module_new_ex and keep it for `wasm_module_new`, since the buffer for the former may be modified, while the later won't. Not sure is it good to you, maybe we can add comments for them?